utils/data_reader.py

```diff
@@ -143,11 +143,17 @@ def get_query(engine, version, args):
             schema=args.v2_schema,
         )
 
-        # the expectation is there will only be one record per individual
-        # in private_address_history, so we simply join the tables
-        # with no further filtering
+        subquery = (
+            select(prv_address.columns.addressid)
+            .filter(prv_address.columns.patid == prv_demo.columns.patid)
+            .order_by(prv_address.columns.address_preferred.desc())
+            .order_by(prv_address.columns.address_period_start.desc())
```

Comment:
```suggestion
            .order_by(prv_address.columns.address_period_start.desc().nulls_last())
```

If this isn't specified it seems to default to "nulls first" which means we get a record with a null start over one with a defined start, which doesn't seem ideal. (In the sample DB Reece set up with examples, see patid 8) Note this is probably not going to happen in practice since the data model IG says implementers should always fill out start date, but good to have it covered just in case.